folium/map.py

```diff
@@ -383,7 +400,11 @@ def __init__(
         self.options = remove_empty(
             draggable=draggable or None, autoPan=draggable or None, **kwargs
         )
+        self.icon: Optional[Icon] = None
         if icon is not None:
+            # here we make sure the icon is rendered after the last marker that needs it
```

Comment:
It feels more natural to make SetIcon a child of Marker and add the Icon to the map instead of to the Marker.